### PR TITLE
feat: add quantile regression to ranger

### PR DESCRIPTION
**Pull request by be-marc**

### CODE REVIEW

File: R/LearnerRegrRanger.R [hunk 1]
```diff
@@ -123,8 +126,23 @@ LearnerRegrRanger = R6Class("LearnerRegrRanger",
...
+        se = "se",
+        quantile = "quantiles"
+      )
+
+      prediction = invoke(predict, self$model, data = newdata, type = type, quantiles = private$.quantile, .args = pv)
+
+      if (type == "quantiles") {
+        response = prediction$predictions[, which(private$.quantile == private$.quantile_response)]
+        quantile = prediction$predictions
+        attr(quantile, "probs") = private$.quantile
```
Comment (mllg) [inline R/LearnerRegrRanger.R:141]: This should be handled by mlr3

File: R/LearnerRegrRanger.R [hunk 2]
```diff
@@ -123,8 +126,23 @@ LearnerRegrRanger = R6Class("LearnerRegrRanger",
       pv = self$param_set$get_values(tags = "predict")
       newdata = ordered_features(task, self)
 
-      prediction = invoke(predict, self$model, data = newdata, type = self$predict_type, .args = pv)
-      list(response = prediction$predictions, se = prediction$se)
+      type = switch(
```
Comment (mllg) [inline R/LearnerRegrRanger.R:129]: I don't get what this is for, this seems unnecessary complicated.

File: tests/testthat/test_regr_ranger.R [hunk 3]
```diff
@@ -66,3 +66,17 @@ test_that("default_values", {
...
+  learner = mlr3::lrn("regr.ranger", num.trees = 100, predict_type = "quantile")
+  learner$quantiles = c(0.1, 0.5, 0.9)
+  learner$quantile_response = 0.5
+  task = tsk("mtcars")
+
+  learner$train(task)
+  pred = learner$predict(task)
+  expect_matrix(pred$quantile)
+  tab = as.data.table(pred)
+  expect_names(names(tab), must.include = c("q0.1", "q0.5", "q0.9", "response"))
```
Comment (mllg) [inline tests/testthat/test_regr_ranger.R:80]: Can I also get the mean and median?

Comment (mllg) [reply tests/testthat/test_regr_ranger.R:80]: The order here is not arbitrary, the quantiles should be sorted in increasing order.

File: R/LearnerRegrRanger.R (hunk 1 above)
Comment (be-marc) [reply R/LearnerRegrRanger.R:141]: Yes. I will move it to mlr3

File: tests/testthat/test_regr_ranger.R (hunk 3 above)
Comment (be-marc) [reply tests/testthat/test_regr_ranger.R:80]: Bernd says no

Comment (mllg) [reply tests/testthat/test_regr_ranger.R:80]: 🤷‍♂️

**Pull request merged**